tools/agent-ctl/src/client.rs

```diff
@@ -294,25 +295,135 @@ fn client_create_vsock_fd(cid: libc::c_uint, port: u32) -> Result<RawFd> {
+
+    if fields.len() != 2 {
+        return Err(anyhow!("invalid server address URI"));
+    }
+
+    let scheme = fields[0].to_lowercase();
+
+    let fd: RawFd = match scheme.as_str() {
+        // Formats:
+        //
+        // - "unix://absolute-path" (domain socket)
+        //   (example: "unix:///tmp/domain.socket")
+        //
+        // - "unix://@absolute-path" (abstract socket)
+        //   (example: "unix://@/tmp/abstract.socket")
+        //
+        "unix" => {
+            let mut abstract_socket = false;
+
+            let mut path = fields[1].to_string();
+
+            if path.starts_with('@') {
+                abstract_socket = true;
+
+                // Remove the magic abstract-socket request character ('@')
+                // and crucially add a trailing nul terminator (required to
+                // interoperate with the ttrpc crate).
+                path = path[1..].to_string() + &"\x00".to_string();
+            }
+
+            if abstract_socket {
+                let socket_fd = match socket(
+                    AddressFamily::Unix,
+                    SockType::Stream,
+                    SockFlag::empty(),
+                    None,
+                ) {
+                    Ok(s) => s,
+                    Err(e) => return Err(anyhow!("Failed to create Unix Domain socket: {:?}", e)),
+                };
+
+                let unix_addr = match UnixAddr::new_abstract(path.as_bytes()) {
+                    Ok(s) => s,
+                    Err(e) => {
+                        return Err(anyhow!(
+                            "Failed to create Unix Domain abstract socket: {:?}",
+                            e
+                        ))
+                    }
+                };
+
+                let sock_addr = SockAddr::Unix(unix_addr);
+
+                match connect(socket_fd, &sock_addr) {
+                    Ok(_) => (),
+                    Err(e) => {
+                        return Err(anyhow!(
+                            "Failed to connect to Unix Domain abstract socket: {:?}",
+                            e
+                        ))
+                    }
+                };
+
+                socket_fd
+            } else {
+                let stream = match UnixStream::connect(path) {
+                    Ok(s) => s,
+                    Err(e) => {
+                        return Err(anyhow!(
+                            "failed to create named UNIX Domain stream socket: {:?}",
+                            e
+                        ))
+                    }
+                };
+
+                stream.into_raw_fd()
+            }
+        }
+        // Format: "vsock://cid:port"
+        "vsock" => {
+            let addr: Vec<&str> = fields[1].split(':').collect();
+
+            if addr.len() != 2 {
+                return Err(anyhow!("invalid VSOCK server address URI"));
+            }
+
+            let cid: u32 = match addr[0] {
+                "-1" => libc::VMADDR_CID_ANY,
```

Comment:
is it better to  merge those two case into one as:  "-1"|"" => libc::VMADDR_CID_ANY   ?

Reply:
Agreed - fixed! :)